packages/toolkit/src/apm/getApmRegistryPackages.js

```diff
@@ -1,24 +1,19 @@
-import APM from '@aragon/apm'
+import getApm from './apm'
 
 /**
  * Return packages for a given APM registry.
- * `progressHandler` can be called with 2 values:
- * - 1: Fetching APM Registry
- * - 2: Gathering packages in registry
  *
- * @param {Object} web3 web3
  * @param {string} apmRegistryName APM registry name
- * @param {Object} apmOptions APM options
  * @param {function(number)} progressHandler Progress handler
+ * @param  {string} environment Envrionment
  * @returns {void}
  */
 export default async (
-  web3,
   apmRegistryName,
-  apmOptions,
-  progressHandler = () => {}
+  progressHandler = () => {},
```

Comment:
We need to start thinking of a way to improve this. I'd love to move it out of parameters and someway subscribe to it in a completely independent way. So, this function would only accept an environment. Then, you would use an event emitter and an event identifier string to listen to the events in another argument.

Reply:
I like that idea and confirm my struggle so far with the `progressHandler`. Would you mind try to include an issue describing a bit more about how that work with events?